gammapy/datasets/flux_points.py

```diff
@@ -17,6 +17,20 @@
 __all__ = ["FluxPointsDataset"]
 
 
+def _get_reference_model(model, energy_bounds, margin_percent=70):
+    if isinstance(model.spatial_model, TemplateSpatialModel):
+        geom = model.spatial_model.map.geom
```

Comment:
What happens here if the template is defined on a much larger geometry than the actual geom used?

Reply:
I think that's the same for any spatial model the extent of the model could be larger than the mask fit and yet we consider the full model.

Reply:
but I can add the a `mask` argument here and pass it to `to_template_spectral_model`